rhino/src/main/java/org/mozilla/javascript/NativeObject.java

```diff
@@ -45,6 +47,7 @@ public Scriptable construct(Context cx, Scriptable scope, Object[] args) {
                 };
 
         var proto = new NativeObject();
+        proto.setParentScope(s);
```

Comment:
Do you know why you had to do this? It's fine if it's necessary, it's just that I am eternally confused since we use ScriptableObject and the prototype hierarchy to represent objects, and the same class and the "parent" hierarchy to represent scopes in the runtime, and I keep wondering if we should make those separate things, even in implementation, so when I see a prototype with a parent I wonder why!

Reply:
LoL, you hit the nail

it is NECESSARY - i spend 2 long debugging sessions to figure this out (this was more or less the hardest part of this change). The problem is that we are in the early phase of the overall setup of the objects/prototoypes and all that stuff. Without that, the lambda functions will not have the correct prototype (Object instead of Function) because the lookup for the buildin prototypes will fail without having a scope.

I did some thinking about adding a comment here but finally decide to not do it because i think having the details will confuse the default reader of that stuff more than it helps. But i can add some sentences if you like....